tests/test_host_check.py

```diff
@@ -1448,116 +1521,114 @@ def test_v1_disabled_in_kernel_config(self, capsys):
+            == " PASS -- Real-time Group Scheduling (disabled in kernel config)\n"
         )
-        assert success
+        assert result.is_success()
 
     def test_v1_disabled_at_runtime(self, capsys):
         """Test the v1 case where disabled at runtime"""
         with mock.patch(
             "host_check._get_cgroup_version", return_value=1
         ), mock.patch("os.path.exists", return_value=True):
-            success, output = self.perform_check(capsys, read_effects="-1")
+            result, output = self.perform_check(capsys, read_effects="-1")
         assert (
             output
-            == "PASS -- Real-time Group Scheduling (disabled at runtime)\n"
+            == " PASS -- Real-time Group Scheduling (disabled at runtime)\n"
         )
-        assert success
+        assert result.is_success()
 
     def test_v1_read_error(self, capsys):
         """Test the limit being too low."""
         with mock.patch(
             "host_check._get_cgroup_version", return_value=1
         ), mock.patch("os.path.exists", return_value=True):
-            success, output = self.perform_check(
-                capsys, read_effects=Exception
-            )
-        assert output == textwrap.dedent(
+            result, output = self.perform_check(capsys, read_effects=Exception)
+        assert output == pretty_print(
             """\
-            WARN -- Real-time Group Scheduling
-                    Failed to read /proc/sys/kernel/sched_rt_runtime_us, unable to check if
-                    real-time group scheduling is disabled.
-                    Running with real-time group scheduling enabled is not supported.
-                    If real-time group scheduling (RT_GROUP_SCHED) is configured in the kernel,
-                    it is required that this feature is disabled at runtime by adding
-                    'kernel.sched_rt_runtime_us=-1' to /etc/sysctl.conf or in a dedicated conf
-                    file under /etc/sysctl.d/.
-                    For a temporary fix, run:
-                      sysctl -w kernel.sched_rt_runtime_us=-1
+            ERROR -- Real-time Group Scheduling
+                     Failed to read /proc/sys/kernel/sched_rt_runtime_us, unable to check if
+                     real-time group scheduling is disabled.
+                     Running with real-time group scheduling enabled is not supported.
+                     If real-time group scheduling (RT_GROUP_SCHED) is configured in the kernel,
+                     it is required that this feature is disabled at runtime by adding
+                     'kernel.sched_rt_runtime_us=-1' to /etc/sysctl.conf or in a dedicated conf
+                     file under /etc/sysctl.d/.
+                     For a temporary fix, run:
+                       sysctl -w kernel.sched_rt_runtime_us=-1
             """
         )
-        assert not success
+        assert not result.is_success()
```

Comment:
Should check for `OverallCheckState.ERRORS`, not just "not success"

Reply:
But if I change `perform_check()` to return a boolean (`result is OverallCheckState.ALL_SUCCEEDED`) and a string, then wouldn't I just revert this change?